modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i

```diff
@@ -0,0 +1,301 @@
+    function = eta1
+    enable_jit = true
+    derivative_order = 2
+  [../]
+  [./var_dependence2]
+    type = DerivativeParsedMaterial
+    f_name = var_dep2
+    args = 'eta2'
+    function = eta2
+    enable_jit = true
+    derivative_order = 2
+  [../]
+
+  [./eigenstrain1]
+    type = ComputeVariableEigenstrain
+    eigen_base = '0.1 -0.1 0 0 0 0'
+    prefactor = var_dep1
+    args = 'eta1'
+    eigenstrain_name = eigenstrain1
+  [../]
+
+  [./eigenstrain2]
+    type = ComputeVariableEigenstrain
+    eigen_base = '-0.1 0.1 0 0 0 0'
+    prefactor = var_dep2
+    args = 'eta2'
+    eigenstrain_name = eigenstrain2
+  [../]
+
+  [./strain]
+    type = ComputeSmallStrain
+    displacements = 'disp_x disp_y'
+    eigenstrain_names = 'eigenstrain1 eigenstrain2'
+  [../]
+
+  [./elastic_free_energy]
+    type = ElasticEnergyMaterial
+    f_name = Fe
+    args = 'eta1 eta2'
+    derivative_order = 2
+  [../]
+
+  [./totol_free_energy]
+    type = DerivativeSumMaterial
+    f_name = F
+    sum_materials = 'Fc Fe'
+    args = 'eta1 eta2'
+    derivative_order = 2
+  [../]
+[]
+
+[AuxVariables]
+  [./sigma11_aux]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./sigma12_aux]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+[AuxKernels]
+  [./matl_sigma11]
+    type = RankTwoAux
+    rank_two_tensor = stress
+    index_i = 0
+    index_j = 0
+    variable = sigma11_aux
+  [../]
+  [./matl_sigma22]
+    type = RankTwoAux
+    rank_two_tensor = stress
+    index_i = 0
+    index_j = 1
+    variable = sigma12_aux
+    execute_on = 'timestep_end initial'
+  [../]
+[]
+
+[BCs]
+  [./bottom_y]
+    type = PresetBC
+    variable = disp_y
+    boundary = 'bottom'
+    value = 0
+  [../]
+  [./top_y]
+    type = PresetBC
+    variable = disp_y
+    boundary = 'top'
+    value = 0
+  [../]
+  [./left_y]
+    type = PresetBC
+    variable = disp_y
+    boundary = 'left'
+    value = 0
+  [../]
+  [./right_y]
+      type = PresetBC
+      variable = disp_y
+      boundary = 'right'
+      value = 0
+  [../]
+  [./top_x]
+    type = PresetBC
+    variable = disp_x
+    boundary = 'top'
+    value = 0
+  [../]
+  [./bottom_x]
+      type = PresetBC
+      variable = disp_x
+      boundary = 'bottom'
+      value = 0
+    [../]
+      [./right_x]
+    type = PresetBC
+    variable = disp_x
+    boundary = 'right'
+    value = 0
+  [../]
+  [./left_x]
+    type = PresetBC
+    variable = disp_x
+    boundary = 'left'
+    value = 0
+  [../]
+[]
+
+[Preconditioning]
+  # active = ' '
+  [./SMP]
+    type = SMP
+    full = true
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  scheme = bdf2
+
+  # this gives best performance on 4 cores
+  solve_type = 'PJFNK'
+  petsc_options_iname = '-pc_type  -sub_pc_type '
+  petsc_options_value = 'asm       lu'
+
+  l_max_its = 30
+  nl_max_its = 10
+  l_tol = 1.0e-4
+  nl_rel_tol = 1.0e-8
+  nl_abs_tol = 1.0e-10
+  start_time = 0.0
+  num_steps = 10
+
+  [./TimeStepper]
+    type = SolutionTimeAdaptiveDT
```

Comment:
@dschwen 
Thanks for your review. It has been updated according to your comments. (except the IterationAdaptiveDT, of which the convergence is much worse than as SolutionTimeAdaptiveDT, seems dt is greatly enlarged after each step, leading to non-convergence)

I will also revise other input file according to this.